components/support/error/src/reporting.rs

```diff
@@ -53,6 +53,10 @@ pub fn set_application_error_reporter(reporter: Box<dyn ApplicationErrorReporter
     *APPLICATION_ERROR_REPORTER.write() = reporter;
 }
 
+pub fn unset_application_error_reporter() {
```

Comment:
Since for a-s rust devs, this feels a tad bit unconventional from the other ways we handle breaking up reference cycles. Maybe adding a comment here referencing "this should be called on desktop to handle ref cleanups/due to callback interfaces" or something to that tune.